clients/client-python/gravitino/filesystem/gvfs_base_operations.py

```diff
@@ -0,0 +1,497 @@
+)
+from gravitino.filesystem.gvfs_config import GVFSConfig
+from gravitino.filesystem.gvfs_storage_handler import get_storage_handler_by_path
+from gravitino.filesystem.gvfs_utils import (
+    get_sub_path_from_virtual_path,
+    extract_identifier,
+    create_client,
+)
+from gravitino.name_identifier import NameIdentifier
+
+logger = logging.getLogger(__name__)
+
+PROTOCOL_NAME = "gvfs"
+
+TIME_WITHOUT_EXPIRATION = sys.maxsize
+
+
+class FilesetPathNotFoundError(FileNotFoundError):
+    """Exception raised when the catalog, schema or fileset is not found in the GVFS path."""
+
+
+class BaseGVFSOperations(ABC):
+    """
+    Abstract base class for Gravitino Virtual File System operations.
+
+    This class provides the core functionality for interacting with the Gravitino Virtual File
+    System, including operations for file manipulation, directory management, and credential
+    handling. It handles the mapping between virtual paths in the Gravitino namespace and actual file
+    paths in underlying storage systems.
+
+    Implementations of this class should provide the specific behaviors for each file system
+    operation.
+    """
+
+    # pylint: disable=too-many-instance-attributes
+
+    SLASH = "/"
+
+    ENABLE_CREDENTIAL_VENDING_DEFAULT = False
+
+    ENV_CURRENT_LOCATION_NAME_ENV_VAR_DEFAULT = "CURRENT_LOCATION_NAME"
+
+    def __init__(
+        self,
+        server_uri: str = None,
+        metalake_name: str = None,
+        options: Dict = None,
+        **kwargs,
+    ):
+        self._metalake = metalake_name
+        self._options = options
+
+        self._client = create_client(options, server_uri, metalake_name)
+
+        cache_size = (
+            GVFSConfig.DEFAULT_CACHE_SIZE
+            if options is None
+            else options.get(GVFSConfig.CACHE_SIZE, GVFSConfig.DEFAULT_CACHE_SIZE)
+        )
+        cache_expired_time = (
+            GVFSConfig.DEFAULT_CACHE_EXPIRED_TIME
+            if options is None
+            else options.get(
+                GVFSConfig.CACHE_EXPIRED_TIME, GVFSConfig.DEFAULT_CACHE_EXPIRED_TIME
+            )
+        )
+        self._filesystem_cache = TTLCache(maxsize=cache_size, ttl=cache_expired_time)
+        self._cache_lock = rwlock.RWLockFair()
+
+        self._catalog_cache = LRUCache(maxsize=100)
+        self._catalog_cache_lock = rwlock.RWLockFair()
+
+        self._current_location_name = self._init_current_location_name()
+
+    @property
+    def current_location_name(self):
+        return self._current_location_name
+
+    @abstractmethod
+    def ls(self, path, detail=True, **kwargs):
+        """List the files and directories info of the path.
+        :param path: Virtual fileset path
+        :param detail: Whether to show the details for the files and directories info
+        :param kwargs: Extra args
+        :return If details is true, returns a list of file info dicts, else returns a list of file paths
+        """
+        pass
```

Comment:
IIRC, we don't have to add `pass` for abstractmethod.

Reply:
No,  without `pass` will lead to syntaxError